src/data/nat/factorial/double_factorial.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2023 Jake Levinson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jake Levinson
+-/
+import data.nat.factorial.basic
+import algebra.big_operators.order
+import tactic.ring
+/-!
+# Double factorials
+
+This file defines the double factorial,
+  `n‼ := n * (n - 2) * (n - 4) * ...`.
+
+## Main declarations
+
+* `nat.double_factorial`: The double factorial.
+-/
+
+open_locale nat
+namespace nat
+
+/-- `nat.double_factorial n` is the double factorial of `n`. -/
+@[simp] def double_factorial : ℕ → ℕ
+| 0       := 1
+| 1       := 1
+| (k + 2) := (k + 2) * double_factorial k
+
+-- This notation is `\!!` not two !'s
+localized "notation (name := nat.double_factorial) n `‼`:10000 := nat.double_factorial n" in nat
+
+lemma double_factorial_add_two (n : ℕ) : (n + 2)‼ = (n + 2) * n‼ := rfl
+
+lemma double_factorial_def' (n : ℕ) : (n + 1)‼ = (n + 1) * (n - 1)‼ := by { cases n; refl }
```

Comment:
```suggestion
lemma double_factorial_add_one (n : ℕ) : (n + 1)‼ = (n + 1) * (n - 1)‼ := by { cases n; refl }
```
I think `def` would be the one with `n - 2`

Reply:
I don't think there can be such a lemma, `n‼ = n * (n-2)‼` is false for `n = 0`.